x-pack/plugins/security_solution/scripts/endpoint/endpoint_agent_runner/index.ts

```diff
@@ -62,6 +65,8 @@ export const cli = () => {
         --username          Optional. User name to be used for auth against elasticsearch and
                             kibana (Default: elastic).
         --password          Optional. Password associated with the username (Default: changeme)
+        --apiKey            Optional. A Kibana API key to use for authz. When defined, 'username'
```

Comment:
Q: I don't see code change that ignores username/password, or is it already baked in?

Reply:
@ashokaditya - yes, its backed into `KbnClientExtended` class above. When api key is defined, we remove the creds from the URL and also intercept the `.request()` calls and inject the API key header